pymc3/distributions/discrete.py

```diff
@@ -438,17 +441,17 @@ class DiscreteUniform(Discrete):
 
     def __init__(self, lower, upper, *args, **kwargs):
         super(DiscreteUniform, self).__init__(*args, **kwargs)
-        self.lower = tt.floor(lower).astype('int32')
-        self.upper = tt.floor(upper).astype('int32')
+        self.lower = intX(np.floor(lower))
```

Comment:
Yeah theano tensor could be distribution bound.

In this case can you not do something similar to `Uniform(Continuous)`:
```python
        self.lower = floatX(lower)
        self.upper = floatX(upper)
```